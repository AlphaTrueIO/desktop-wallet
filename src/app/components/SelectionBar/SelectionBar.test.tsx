import { fireEvent, render } from "@testing-library/react";
import { act, renderHook } from "@testing-library/react-hooks";
import React from "react";

import { SelectionBar, SelectionBarGroup, useRadioState } from "./index";

describe("SelectionBarGroup", () => {
	it("should render", () => {
		const { getByTestId, asFragment } = render(<SelectionBarGroup />);
		expect(getByTestId("SelectionBarGroup")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});
});

describe("SelectionBar", () => {
	it("should render", () => {
		const isValueChecked = jest.fn((value: any) => (value === 1 ? true : false));
		const setCheckedValue = jest.fn();
		const { getAllByTestId, asFragment } = render(
			<>
				<SelectionBar isValueChecked={isValueChecked} setCheckedValue={setCheckedValue} value={1}>
					Test 1
				</SelectionBar>
				<SelectionBar isValueChecked={isValueChecked} setCheckedValue={setCheckedValue} value={2}>
					Test 2
				</SelectionBar>
				,
			</>,
		);

		const buttons = getAllByTestId("SelectionBar");
		expect(buttons[0]).toHaveAttribute("aria-checked", "true");
		expect(buttons[1]).toHaveAttribute("aria-checked", "false");

		fireEvent.click(buttons[0]);
		expect(setCheckedValue).toHaveBeenCalledWith(1);

		expect(asFragment()).toMatchSnapshot();
	});

	it("should work with useRadioState", () => {
		const { result: state } = renderHook(() => useRadioState(undefined));
		const { getAllByTestId } = render(
			<>
				<SelectionBar {...state.current} value={1}>
					Test 1
				</SelectionBar>
				<SelectionBar {...state.current} value={2}>
					Test 2
				</SelectionBar>
				,
			</>,
		);

		const buttons = getAllByTestId("SelectionBar");
		expect(buttons[0]).toHaveAttribute("aria-checked", "false");
		expect(buttons[1]).toHaveAttribute("aria-checked", "false");

		act(() => {
			fireEvent.click(buttons[0]);
		});

		expect(state.current.checkedValue).toEqual(1);
	});
});
