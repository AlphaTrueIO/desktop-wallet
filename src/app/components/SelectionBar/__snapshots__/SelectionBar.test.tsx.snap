// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionBar should render 1`] = `
<DocumentFragment>
  <div
    class="border-theme-neutral-300 inline-flex items-center flex-shrink-0 overflow-hidden border rounded shadow-sm"
    data-testid="SelectionBar"
    role="radiogroup"
  />
</DocumentFragment>
`;

exports[`SelectionBarOption should render 1`] = `
<DocumentFragment>
  <button
    aria-checked="true"
    class="sc-AxjAm dlugZr focus:outline-none border-b-3 relative px-5 py-3 transition-colors duration-300 border-transparent"
    data-testid="SelectionBarOption"
    role="radio"
  >
    Test 1
  </button>
  <button
    aria-checked="false"
    class="sc-AxjAm dlugZr focus:outline-none border-b-3 relative px-5 py-3 transition-colors duration-300 border-transparent"
    data-testid="SelectionBarOption"
    role="radio"
  >
    Test 2
  </button>
  ,
</DocumentFragment>
`;
